File: api/src/main/java/org/pmiops/workbench/actionaudit/targetproperties/PreviousNewValuePair.java
package org.pmiops.workbench.actionaudit.targetproperties;

public record PreviousNewValuePair(String previousValue, String newValue) {
  public boolean hasValueChanged() {
    return (previousValue == null && newValue != null)
        || (previousValue != null && !previousValue.equals(newValue));
  }
}

File: api/src/main/java/org/pmiops/workbench/actionaudit/targetproperties/PreviousNewValuePair.kt
package org.pmiops.workbench.actionaudit.targetproperties

data class PreviousNewValuePair(var previousValue: String?, var newValue: String?) {

    val valueChanged: Boolean
        get() {
            return previousValue == null && newValue != null ||
                    newValue == null && previousValue != null ||
                    previousValue != newValue
        }
}
